library/tiqr/Tiqr/Message/FCM.php

```diff
@@ -62,15 +60,20 @@ private function getGoogleAccessToken($credentialsFile, $cacheTokens, $tokenCach
             $pool = new FilesystemCachePool($filesystem);
 
             //set up a callback to log token refresh
-            $tokenCallback = function ($cacheKey, $accessToken) {
-                $this->logger->debug(sprintf('New access token received at cache key %s', $cacheKey));
+            $logger=$this->logger;
+            $tokenCallback = function ($cacheKey, $accessToken) use ($logger) {
+                $logger->info(sprintf('New access token received at cache key %s', $cacheKey));
             };
             $client->setTokenCallback($tokenCallback);
             $client->setCache($pool);
         } else {
             $this->logger->warning("Cache for oAuth tokens is disabled");
         }
-        $client->setAuthConfig($credentialsFile);
+        try {
+            $client->setAuthConfig($credentialsFile);
+        } catch (\Google\Exception $e) {
+            throw new Tiqr_Message_Exception_SendFailure(sprintf("Error setting Google credentials for FCM : %s", $e->getMessage()), true);
+        }
```

Comment:
When throwing an exception you can add the previous exception as a reference to the exception. You now just put the message, which is probably enough. 

throw new FoobarException('New message with the old %s', $code, $e);

The thire param usually is the `previous` throwable. 

Just saying.

Reply:
Thanks for the suggestion, I've extended the exception